securedrop_client/gui/actions.py

```diff
@@ -23,7 +23,7 @@ def __init__(
     ) -> None:
         self._controller = controller
         self._state = app_state
-        self._text = _("All Files")
+        self._text = _("Download All Files")
```

Comment:
I'll bring @tina-ux to this conversation (no pun intended)!
I think the current versions have the benefit of being unambiguous (e.g Will "Download Files" give me a chance to select which files? seems like a fair expectation to me.) and @tina-ux and I talked a bit about that. But I don't have a strong preferrence and it's easy to change if journalists tell us that confused them!

Regarding the "Conversation" word, I had doubts introducing the name in this PR because it's not one we've used before. As you know, that's how I think of the collection of files and messages, but I think @tina-ux would prefer having some time to think about different analogies we could use before comitting to "conversation". Enough of me, over to you @tina-ux!

Reply:
Just to elaborate more on why I would suggest this...

I use this menu often, and one of the things I've noticed is that I feel like I might be choosing the wrong option, even after rereading the options once or twice. I think it has to do with there not being an icon or a keyword that reassures me that it IS the option I'm hunting for. The consequence of selecting the wrong option is just that you may open the wrong dialog, so I don't have super strong feelings about it. I just know that we don't have a lot of people using this feature, so figured I'd leave my two cents. As you can see in the original prototype, we did envision introducing icons: https://github.com/freedomofpress/securedrop-client/issues/1354

Reply:
See https://user-images.githubusercontent.com/8262612/143721790-a3500b24-d7e2-46df-a12f-893130a86674.png

Reply:
Hey folks, when we talked about it with Gonzalo, I did indeed opt for more verbose but unambiguous menu action strings. Overall, I feel like I don't know enough about how journalists think about the workstation to be able to abbreviate.

I can break down my thinking for each of those:

- **"Download All Files" vs "Download Files":** Without "All", it is not clear that every single file within the thread/conversation will be downloaded. "Download Files" on its own begs the question "which files exactly?" especially in a longer conversation. Like Gonzalo said, if I clicked on Download Files, I would expect a selection process first, whereas download all files is hard for me to interpret any other way.

- **"Delete All Files and Messages" vs "Delete Conversation"**: Two things come to mind: 
a) The easier issue to address: there is some ambiguity there because we haven't used the term "conversation" before (like G mentioned) and while I am absolutely not opposed to introducing this new term, it does beg the question: which mental model are we settling on and opting to communicate? Is this a messaging thread (which emphasizes the asynchronous nature of the exchange), is this a conversation (more instant)?  This part I think we can settle on fairly quickly even in this PR (and worst case we choose a different word down the line), but then there is a more basic question which is: 
b) when we tell a user that they can delete the conversation, what do we mean exactly and how do we convey this with precision? As a user, am I simply deleting the content of the conversation (messages and files) or am I deleting the whole thread (and won't be able to find the name of the source anymore in my list after I do this)? Again, deleting an email thread means deleting the item in your inbox as well, however, deleting messaging conversation on signal or whatsapp means deleting the content of the conversation SO FAR, with the option to select the recipient from a list and start a brand new conversation with them. It also means I am deleting the conversation for myself only.  In SD, because sources are anonymized and they don't live in a contact list outside of the inbox and the threads, we don't follow that logic if I'm not mistaken. In order to delete the whole thread, journalists are instructed to delete the source. But then... why is it that that means deleting the source _account_ (meaning that I am deleting the credentials of my source). As a journalist, why am I acting as an admin and removing agency from the source? (if it's to manage spam and unsafe threads, then maybe there are other solutions to look into). And if I am indeed deleting the source _account_ then I should be acutely aware of it since it can have such a big impact on the source. This leads to us to a more fundamental line of questioning: why can't a source delete their own account and should a journalist be able to delete their account without their consent? So I feel for this menu item and the "delete source" it's hard for me to abbreviate as long as we have not had the resources to think these interactions through and not have them bleed into each other.  

I can't wait until there is enough preliminary work done on my end but also until my plate is free enough so I can dedicate some serious time to the workstation.

I will say that i would like to have the "delete" actions be red, or highlighted since they are high risk actions (that cannot be reverted), and they should require an explicit confirmation after they are selected (which I think is already the case right now). 

Apologies for the block of text! I am happy to have a synchronous convo about this as well

Reply:
Makes sense @tina-ux. Seems like colors, bolding, and icons would be a better way to give visual cues as to what kind of option you're selecting rather than truncating the option text.

Reply:
> Just to elaborate more on why I would suggest this...
> 
> I use this menu often, and one of the things I've noticed is that I feel like I might be choosing the wrong option, even after rereading the options once or twice. I think it has to do with there not being an icon or a keyword that reassures me that it IS the option I'm hunting for. The consequence of selecting the wrong option is just that you may open the wrong dialog, so I don't have super strong feelings about it. I just know that we don't have a lot of people using this feature, so figured I'd leave my two cents. As you can see in the original prototype, we did envision introducing icons: #1354

I hear you! Then yes it may be a question of making the action items visually distinct as well / sorting them visually while keeping the verb in the string. Like I said I'd like to make the Delete actions very distinguishable because they carry the most risk and it is increasingly standard to do so. I don't have enough time this week to dive into prototypes but I am happy to figure out a timeline together

Reply:
I was about to mention that the highlight/selection color could be made red for the destructive actions (as well as the default text color if we wanted to? I'm less sure oersonally about that, because it also unbalances the menu items IMHO) but I'd prefer addressing that in a separate piece/PR to keep scopes tight.

Also, and maybe that's something I's do in the scope of this PR, note that most Linux menu entries that do not have immediate effect (e.g. because they open a dislog where you can change your mind and cancel) have their namesyse suspension dots. Example: "Download All Files" but "Download Files..." (if we were to offer selection options) or "Delete Conversation..." (that requires confirmation). That slipped from my mind when we paired last week @tina-ux.

Reply:
Before I open a specific issue to discuss / track this, let me try to refine a little bit on what I think we're talking about (please correct me if I miss anything / get anything wrong!):

I see two problems being raised in this thread:
- How do I know which actions are potentially **irreversible**?
- How do I know which actions will take place **immediately** after I trigger the menu entry?

They are related, but I think distinct problems. Sometimes (most of the times?) we'd likely try not to make irreversible actions also immediate, to prevent accidents. But I also think we can address them independently in the menu. (And I'm talking about menus specifically.)

Example solutions that come to my mind:
- Immediate vs "explorable": the `...` naming convention that I mentioned in the comment.
- Irreversible: when hovered/selected could have a different background color (e.g. red instead of blue).

Both obviously can be true at the same time/work together, which is something we'd want to ensure if we treat them separately. :slightly_smiling_face: 

As a more general guideline, I think that trying to give options to people to discover their options by exploring and providing ways to undo/cancel is a useful conversation starter at least (from previous experience).

<details><summary>Example screenshots (but they're examples only, that's why I hide them :wink:</summary>
Menu items with suspension dots indicate a dialog will open (and I'll be able to change my mind and "cancel"):

<img src="https://user-images.githubusercontent.com/1619067/172264892-58911362-7c5f-455f-a9ef-26ed4afddfed.png" alt="menu items with suspension dots indicate a dialog will open"/>

Regular (non-destructive, or reversible) options are highlighted as usual:

<img src="https://user-images.githubusercontent.com/1619067/172264896-5d97e779-565b-4625-a3eb-32829327614a.png" alt="Regular (non-destructive, or reversible) options are highlighted as usual"/>

Irreversible actions are highlighted in a distinctive way:

<img src="https://user-images.githubusercontent.com/1619067/172264897-2e648a4c-abca-4a00-893c-6e8f6f310d9d.png" alt="Irreversible actions are highlighted in a distinctive way."/>

</details>